packages/extension/src/inpage/alephiumWindowObject.ts

```diff
@@ -207,19 +197,67 @@ export const alephiumWindowObject: AlephiumWindowObject = new (class extends Ale
       throw Error("User action timed out")
     }
 
-    console.log(`==== result`, result)
     return result
   }
 
-  protected unsafeGetSelectedAccount(): Promise<Account> {
-    if (this.connectedAccount) {
-      return Promise.resolve(this.connectedAccount)
-    } else {
-      throw Error("No selected address")
+  async disconnect(): Promise<void> {
+    this.#connectedAccount = undefined
+    this.#connectedNetworkId = undefined
+    this.#nodeProvider = undefined
+    this.#explorerProvider = undefined
+    await removePreAuthorization()
+    if (this.onDisconnected !== undefined) {
+      await this.onDisconnected()
+    }
+    this.onDisconnected = undefined
+  }
+
+  #checkParams({ signerAddress }: { signerAddress: Address }): void {
+    if (this.#connectedAccount === undefined || this.#connectedNetworkId === undefined) {
+      throw Error("No connection")
+    }
+
+    if (signerAddress !== this.connectedAccount?.address) {
+      throw Error("Unauthorized address")
```

Comment:
[Nit] Maybe we can print the two addresses in the exception for ease of debugging.